src/main/java/org/jenkinsci/plugins/KeycloakAccess.java

```diff
@@ -48,14 +48,18 @@ public KeycloakAccess( KeycloakDeployment keycloakDeployment ) {
 	public UserDetails loadUserByUsername( String username ) throws UsernameNotFoundException, DataAccessException {
 		LOGGER.finer( "Requested User Details for: " + username );
 		try {
-			String token = getAuthToken();
-			List<GrantedAuthority> authorities = getAuthorities(getRolesForUser( username, token ));
-			return new KeycloakUserDetails( username, authorities.toArray( new GrantedAuthority[0] ) );
+			if (!cache.isInvalidUser( username )) {

```

Comment:
If a user hasn't logged in yet would this code still check to see if they are an actual user or would it immediately fail and then they would be marked as an invalid user (since they wouldn't be in the usermap yet)?